agent/pkg/controllers/entries_controller.go

```diff
@@ -36,6 +38,58 @@ func Error(c *gin.Context, err error) bool {
 	return false // no error, can continue
 }
 
+func GetEntries(c *gin.Context) {
+	entriesRequest := &models.EntriesRequest{}
+
+	if err := c.BindQuery(entriesRequest); err != nil {
+		c.JSON(http.StatusBadRequest, err)
+	}
+	validationError := validation.Validate(entriesRequest)
+	if validationError != nil {
+		c.JSON(http.StatusBadRequest, validationError)
+	}
+
+	data, meta, err := basenine.Fetch(shared.BasenineHost, shared.BaseninePort,
+		entriesRequest.LeftOff, entriesRequest.Direction, entriesRequest.Query,
+		entriesRequest.Limit, time.Duration(entriesRequest.TimeoutMs)*time.Millisecond)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, validationError)
+	}
+
+	result := make(map[string]interface{})
+	var dataSlice []interface{}
+
+	for _, row := range data {
+		var dataMap map[string]interface{}
+		err = json.Unmarshal(row, &dataMap)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"error":     true,
+				"type":      "error",
+				"autoClose": "5000",
+				"msg":       string(row),
+			})
+			return // exit
+		}
+
+		base := dataMap["base"].(map[string]interface{})
+		base["id"] = uint(dataMap["id"].(float64))
+
+		dataSlice = append(dataSlice, base)
+	}
+
+	var metadata *basenine.Metadata
+	err = json.Unmarshal(meta, &metadata)
+	if err != nil {
+		logger.Log.Debugf("Error recieving metadata: %v", err.Error())
+	}
+
+	result["data"] = dataSlice
+	result["meta"] = metadata
```

Comment:
I'm not sure the meta will be important but it is not harming to add metadata instead of iterating over the response 
 - Nice change.
 
The reason I wrote about it is that "GetEntries" not really returning array of entries.
Anyway this is a good change

Reply:
Maybe we can just create struct instead of `[]interface{}` and `map[string]interface{}`

Reply:
Never return a JSON array. Instead return a JSON object that contains a JSON array. See;
- https://softwareengineering.stackexchange.com/a/327632/236831
- https://stackoverflow.com/questions/3503102/what-are-top-level-json-arrays-and-why-are-they-a-security-risk

Reply:
@mertyildiran 
Maybe I didn't explain myself, I think that changing to object that holds "meta" and "result" (which is the really json array is OK. 
I just think that we can create struct 
```type EntriesResponse struct {
            Data map[string]interfact{}  /// if we can specify exact type here it is even better
            Meta *basenine.Metadata
```
and then return this, so when someone wants to make change or read this function it will be easier to understand what is this function returning.. 

Again, it is not blocker and I'm ok with current code - just think it might be better and want to know if there is any reason not to do it like this..

Reply:
@gadotroee I think https://github.com/up9inc/mizu/commit/2c263db7b9c7a306ca3ab968917e379c6f40f27a meets your request.